From f0ca76a81b1a1da2ef562ea8a0646d70f4df6704 Mon Sep 17 00:00:00 2001
From: Sam Ravnborg <sam@ravnborg.org>
Date: Tue, 7 Aug 2018 22:53:03 +0200
Subject: [PATCH 609/610] skovarm9cpu: add reset-gpios to macb to fix network

The Davicom PHY on the ARM9 CPU require a reset cycle
to work properly.
Use the reset-gpios property to to reset the
phy during early probe.
This fixes that we can obtain IP address using DHCP.

Signed-off-by: Sam Ravnborg <sam@ravnborg.org>
---
 arch/arm/dts/skovarm9cpu.dts | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm/dts/skovarm9cpu.dts b/arch/arm/dts/skovarm9cpu.dts
index 72f95fb0b..7e127ee2c 100644
--- a/arch/arm/dts/skovarm9cpu.dts
+++ b/arch/arm/dts/skovarm9cpu.dts
@@ -52,14 +52,16 @@
 
 			macb0: ethernet@fffbc000 {
 				phy-mode = "rmii";
+				#address-cells = <1>;
+				#size-cells = <0>;
 				status = "okay";
-/*			
-				ethernet-phy@0 {
-					reg = <0x1>;
-					interrupt-parent = <&pioE>;
-					interrupts = <31 IRQ_TYPE_LEVEL_LOW>;
+
+				ethphy0: ethernet-phy@1 {
+					reg = <3>;
+					reset-gpios = <&pioE 17 GPIO_ACTIVE_LOW>;
+					reset-assert-us = <1000>;
+					reset-deassert-us = <100>;
 				};
-*/
 			};
 
 			mmc1: mmc@fff84000 {
-- 
2.12.0

